### ISSUE TITLE
Opening KDB database results in error "unknown datafield. Password wrong? (1009)" as in #632 since update to 0.8.1

### ISSUE THREAD

**Issue by sebastian-muthwill** [MacPass/MacPass#1221]
**Describe the bug**
Since the update to version 0.8.1 I can no longer open my kdb vault. 
The database is a kdb database. It still can be opened with Keepass2Android on my mobile.

Algorithmus: AES/Rijndael (256-bit key, FIPS 197)
with password and key file (both are korrekt!)

A downgrade to 0.8.0 will result in the same error with 0.8.0. Bevor the update to 0.8.1 it worked with 0.8.0.

The same bug was already reported in #632 with version 0.4.3.

**To Reproduce**
1. enter password
2. see error "unknown datafield" (in german: "Unbekanntes Datenfeld. Passwort Falsch? (1009)

**Expected behavior**
Database can be opened

**Screenshots**
![image](https://user-images.githubusercontent.com/10375725/157848312-d66e439e-3e31-4a52-8ba1-ea534f30707b.png)

**Desktop (please complete the following information):**
 - macOS: [e.g. 12.0.1]
 - Version [e.g. 0.8.1 / 0.8.0]

**Comment by lislon** [comment]
I have same issue with 0.7.12-0.8.1 versions when using following configuration:

 - keepass2Android 1.09c-r0
 - .kdb file is hosted on OneDrive share
 - keyfile+password is used
 - macpass version (0.7.11, 0.7.12, 0.8.1)

👀 One observation: 
When I copy kdb to same directory:
```
cp ~/OneDrive/keepass.kdb ~/OneDrive/keepass-copy.kdb
```
 macpass 0.7.11 opens keepass-copy.kdb file succesfully (but not 0.8.1 and 0.8.0)

**Comment by mrwonko** [comment]
I share my kdb file between KeePass on Windows and MacPass and have started getting "Unknown file format (1039)" errors when I try to unlock in MacPass 0.8.1 on macOS 12.3. I'm not sure if that has the same cause as the issue described above, or is due to me starting to use custom icons for my groups and entries in KeePass on Windows.

**Comment by mstarke** [comment]
@sebastian-muthwill @mrwonko do you use keyfiles?

**Comment by sebastian-muthwill** [comment]
Yes both key file and password in combination.

**Comment by mstarke** [comment]
Hm. That might be the problem. I’ll take a look. There where some changes to the key file generation and those might have introduced regressions.

**Comment by mrwonko** [comment]
I tried downgrading, but neither 0.8 nor 0.7.12 were able to open my db.

Then I tried lislon's trick of copying the database, and I was able to open the copied version, including in 0.8.1. Just renaming the file did not help, and somehow MacPass was aware of the renaming despite being shut down at the time.

I now use a copy of my original database, renamed back to its original name, which for some reason fixed the issue.

Like lislon, I store my db on OneDrive. Maybe that's somehow related?

Meanwhile, my earlier suspicion that this was somehow related to the use of custom icons has been disproven, the custom icons work just fine. 👍

**Comment by DominusExult** [comment]
Yes, that's a problem with how OneDrive works these days. Your database was not on your real drive but only something similar to a symlink. So MacPass tried to open an enpty file (or whatever is represented by OneDrive).
I think I've posted a link hinting at this in another issue.
